### DISCUSSION TITLE
A new feature is needed that allows me to specify the port when setting the WOL address

### DISCUSSION THREAD

**Discussion by commandoccz** [seriousm4x/UpSnap#450] [Feature requests]
Just like this:
![WOL port](https://github.com/seriousm4x/UpSnap/assets/76011848/0d3ac1cc-18e6-4d6c-a2bf-a616025b1bf2)

If it is possible to specify the port when setting the WOL address, then I will be able to wake up the computer across subnets by forwarding the port through the router.

**Comment by seriousm4x** [comment]
Can't you just forward the default port? Port 9.

**Reply by commandoccz** [reply]
Okay，I originally thought that ports lower than 1024 could not be forwarded, but I just tested it and it can be forwarded normally.
![image](https://github.com/seriousm4x/UpSnap/assets/76011848/6fc8b26e-13fc-4355-bd8a-87e689181af2)

**Reply by seriousm4x** [reply]
Nice 👍🏼

**Reply by commandoccz** [reply]
But I'd like to see a port setting feature added, as in some cases, there is still a limit to the minimum port that can be forwarded.

**Reply by masterlog80** [reply]
Following this discussion as in some specific scenario like Docker packets on port 9 may not be sent to the target Network.
I believe this can be fixed by modifying Network settings on the Hots, but it may present some challenges if the OS may be updated, so the simplest solution would be to set a port different from 9 but that doesn't seem possible.
Can you please consider to implement this solution?

Here are my tests done by running command from Container's CLI and collecting packets with WireShark: I noticed that using a different port (assuming UpSnap uses port 9) for WOL fixes the issue as below:
```
awake -p 80 -b 192.168.10.255 AA:BB:CC:DD:EE:GG
```
While commands like this has no effects:
```
awake -b 192.168.10.255 AA:BB:CC:DD:EE:GG
```

Regards,
